cpp11/Ice/interceptor/AuthenticatorI.cpp

```diff
@@ -34,21 +34,16 @@ AuthenticatorI::getToken(string username, string password, const Ice::Current&)
         tokenBuilder[i] = chars.at(dist(_rand));
     }
 
-    //
-    // Create a token object, store a copy, and return it.
-    //
-    Demo::Token token;
-    token.value = string(tokenBuilder);
-    token.username = username;
+    string token = string(tokenBuilder);
```

Comment:
This should use the string constructor that takes a length as the second argument, because `tokenBuilder` is not null terminated.

```
    string token(tokenBuilder, 32);